pkg/encrypt/dare.go

```diff
@@ -0,0 +1,87 @@
+/*
+ * Minio Go Library for Amazon S3 Compatible Cloud Storage (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package encrypt
+
+import (
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/base64"
+	"errors"
+	"io"
+	"io/ioutil"
+
+	"github.com/minio/sio"
```

Comment:
This is the first dependency from outside minio/minio-go.. I am not sure, shouldn't we vendorize this ?

Reply:
Unless we are sticking to a version we should vendorize. If we say sio will be always backward compatible we don't have to.

Reply:
Not really. E.g `github.com/dustin/go-humanize`
With sio the situation is better IMO because it's our repository...

Reply:
> Not really. E.g github.com/dustin/go-humanize
> With sio the situation is better IMO because it's our repository...

Well I found go-humanize only in functional_tests.go

Anyway, I don't think this is urgent and needs to be addressed right now, so we can let it until we need to open this subject again.

Reply:
Oh, your right - I just looked into the appveyor.yml...
Agree we can revisit later...